### PR TITLE
[iOS] Display alerts on top of popups

### LINKED ISSUE #16321: DisplayAlert Displaying Beneath PopupPage (Mopups) on iOS
### Description

On iOS, using DisplayAlert from a Mopups PopupPage causes the alert to be displayed under the PopupPage.
[Here's a link](https://github.com/LuckyDucko/Mopups/issues/66) to the Mopups ticket.

### Steps to Reproduce

1. Show a popup page (Mopups)
2. Display an alert

**Expected**
Alert to be shown in front of the popup page.
**Actual**
Alert is shown behind the popup page.

### Link to public reproduction project repository

https://github.com/LuckyDucko/Mopups/issues/66

### Version with bug

7.0.49

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Display alerts on top of any ViewController.

fix-alert-popup-ios

### Issues Fixed

Fixes #16321

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
This didn't work for me when using mopups

Maybe I'm testing it incorrectly?

I've put a repro here using the sandbox
https://github.com/dotnet/maui/tree/fix-16321-repro

You'll have to modify the Nuget.Config file locally in order to restore mopups

- Run sandbox
- click "Push Mopups this will popup the mopups window
- now click "Display Alert", this calls DisplayAlert

If you pop the "Mopups" you'll notice that the alert is under the mopups.


I think the problem is that the "Mopups" are displaying inside a new UIWindow so that's not going to have a relationship you can find via "PresentingViewController" 

One thought here would be to inspect the incoming page and use the platform window that's on that page as the source ViewController

`PresentPopUp` takes a UIWindow argument so you could pass in the UIWindow from the page that `PresentAlert` was called on.

**Comment by jsuarezruiz** [comment]
> This didn't work for me when using mopups
> 
> Maybe I'm testing it incorrectly?
> 
> I've put a repro here using the sandbox https://github.com/dotnet/maui/tree/fix-16321-repro
> 
> You'll have to modify the Nuget.Config file locally in order to restore mopups
> 
> * Run sandbox
> * click "Push Mopups this will popup the mopups window
> * now click "Display Alert", this calls DisplayAlert
> 
> If you pop the "Mopups" you'll notice that the alert is under the mopups.
> 
> I think the problem is that the "Mopups" are displaying inside a new UIWindow so that's not going to have a relationship you can find via "PresentingViewController"
> 
> One thought here would be to inspect the incoming page and use the platform window that's on that page as the source ViewController
> 
> `PresentPopUp` takes a UIWindow argument so you could pass in the UIWindow from the page that `PresentAlert` was called on.

Thanks for testing and the feedback. Let me try again, the previous attached gif was using Mopups (the yellow popup) but maybe missed something.

**Comment by PureWeen** [comment]
Replaced by https://github.com/dotnet/maui/pull/16983
